src/core/cycling/idGenerator.js

```
@@ -42,4 +42,41 @@ export const generateId = {
+      isNaN(numIndex) ||
+      !Number.isInteger(numIndex) ||
+      numIndex < 1
+    ) {
+      throw new Error(
+        `generateId.location: invalid index (${index}) - must be a positive integer`,
+      );
+    }
+
+    if (!locationType || !String(locationType).trim()) {
+      throw new Error(
+        `generateId.location: invalid locationType (${locationType}) - must be one of: ${Array.from(ALLOWED_LOCATION_TYPES).join(", ")}`,
```

ESLint

[error] 'ALLOWED_LOCATION_TYPES' is not defined.

(no-undef)